File: 3457.eat-pizzas.py
#
# @lc app=leetcode id=3457 lang=python3
#
# [3457] Eat Pizzas!
#
# @lc code=start
class Solution:
    def maxWeight(self, pizzas: List[int]) -> int:
        pizzas.sort(reverse=True)  # Step 1: Sort in descending order.
        total_gain = 0
        # Iterate over the sorted pizzas in chunks of 4.
        for i in range(0, len(pizzas), 4):
            if (i // 4) % 2 == 0:  # Step 2: Check if it's an odd day (0-indexed).
                total_gain += pizzas[i]     # Gain Z on odd days (largest)
            else:
                total_gain += pizzas[i + 1] # Gain Y on even days (second largest)
        return total_gain # Step 3: Return total gained weight.
# @lc code=end

File: 3457.eat-pizzas.java
#
# @lc app=leetcode id=3457 lang=java
#
# [3457] Eat Pizzas!
#
# @lc code=start
class Solution {
    public long maxWeight(int[] pizzas) {
        Arrays.sort(pizzas);
        int n = pizzas.length;
        long res = 0;
        int days = n / 4;
        for (int i = 0; i < days; ++i) {
            int idx = n - 4 * i;
            if ((i & 1) == 0) { // odd-numbered day (1-indexed)
                res += pizzas[idx - 1]; // largest (Z)
            } else { // even-numbered day
                res += pizzas[idx - 2]; // second largest (Y)
            }
        }
        return res;
    }
}
# @lc code=end